assessment/collectors/app_migration_analysis.go

```diff
@@ -275,7 +295,13 @@ func (m *MigrationSummarizer) fetchFileContent(filepath string) (string, error)
 	return content, nil
 }
 
-func (m *MigrationSummarizer) AnalyzeFile(ctx context.Context, filepath, methodChanges, content string, fileIndex int) (*CodeAssessment, []any) {
+func (m *MigrationSummarizer) AnalyzeFileChannel(ctx context.Context, filepath, methodChanges, content string, fileIndex int, bufferedChannel chan *AnalyzeFileResponse, wg *sync.WaitGroup) {
```

Comment:
I'm not sure where this wait group is being initialized and where it is being closed. Can you use the runParallelTasks method in utils for parallelism instead ?

Reply:
Done.